### PR TITLE
act: init at 0.1.3

### PR DESCRIPTION
**Pull request by novoxd**
###### Motivation for this change
Add act tool https://github.com/nektos/act .
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

P.S. This is my first package build and first pr to nixpkgs, sorry if something wrong.

### CODE REVIEW

File: pkgs/tools/admin/act/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  subPackages = [ "." ];
+
+  buildFlags = [ "-tags netgo" "-tags release" ];
+
+  #postInstall =
+  #''
+  #  mkdir -p "$out/share/"{bash-completion/completions,zsh/site-functions}
+  #  $out/bin/eksctl completion bash > "$out/share/bash-completion/completions/eksctl"
+  #  $out/bin/eksctl completion zsh > "$out/share/zsh/site-functions/_eksctl"
+  #'';
```
Comment (kalbasit) [inline pkgs/tools/admin/act/default.nix]: Please either implement this or remove the dead code.

**Review by kalbasit — CHANGES_REQUESTED** [review]
The commit message must be `act: init at 0.1.3`. Same as the PR title.

**Comment by novoxudonoser** [comment]
> The commit message must be `act: init at 0.1.3`. Same as the PR title.

Thanks for little guide how to. Pr fixed.

**Comment by Lassulus** [comment]
getting this error on reviewing:
```
hash mismatch in fixed-output derivation '/nix/store/anr7nkf60q3wjj2mjxn4cfidaw2b3skv-act-0.1.3-go-modules':
  wanted: sha256:0g7j9ysvsh7k8sg5nkxmkqnsa0xl824hynfg6cbmdydmb4lvv9rq
  got:    sha256:1c8bx9jb14f70aihnjbzk1js110s9r4nkdfps27kphn65bw2pxn0
cannot build derivation '/nix/store/ny7rk6ckqbrj04p3lqjcyapa9hir2rqd-act-0.1.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/lgws18r6l3hm64zjsky8kaqq5jimgcb9-env.drv': 1 dependencies couldn't be built
```

**Comment by novoxudonoser** [comment]
@Lassulus fixed sha mismatch please recheck.

 @kalbasit as I know you are working with go in nixpkgs (had read your post about buildGoModule). I found some issue while rebuilding this tool, I don't know is it an error or it something wrong in my system. Previously on 18.09 the build was fine and now after upgrade to 19.09 I have an error `go: error loading module requirements`. I googled this issue https://github.com/golang/go/issues/31162 seems like related. Form [stakoverflow link  ](https://stackoverflow.com/questions/55430150/go111module-on-error-loading-module-requirements) looks like some problems in new go version with old packages. I am not so familiar with go and its ecosystem so I cant determine what is the error and where it happened, so need some help.  
```
nix-build -E 'with import <nixpkgs> {}; callPackage ./default.nix {}'
these derivations will be built:
  /nix/store/4ahav8mgks3nmjkpgypvrrbd3isigxg8-act-0.1.3-go-modules.drv
  /nix/store/hnq4f3p1b5fbq20nwkglvgw8v593xja5-act-0.1.3.drv
building '/nix/store/4ahav8mgks3nmjkpgypvrrbd3isigxg8-act-0.1.3-go-modules.drv'...
unpacking sources
unpacking source archive /nix/store/i7sn1mplclhcslx3jr4qbjp4335l80x4-source
source root is source
patching sources
configuring
building
go: finding github.com/opencontainers/image-spec v1.0.1
go: finding github.com/actions/workflow-parser v1.0.0
...
go: finding google.golang.org/genproto v0.0.0-20190128161407-8ac453e89fca
go: error loading module requirements
builder for '/nix/store/4ahav8mgks3nmjkpgypvrrbd3isigxg8-act-0.1.3-go-modules.drv' failed with exit code 1
cannot build derivation '/nix/store/hnq4f3p1b5fbq20nwkglvgw8v593xja5-act-0.1.3.drv': 1 dependencies couldn't be built
error: build of '/nix/store/hnq4f3p1b5fbq20nwkglvgw8v593xja5-act-0.1.3.drv' failed

```

**Comment by Lassulus** [comment]
hmm, seems a little cursed here
```
hash mismatch in fixed-output derivation '/nix/store/0s7bi65pmgf5dzyn5iapa3mgmm8z89kc-act-0.1.3-go-modules':
  wanted: sha256:1c8bx9jb14f70aihnjbzk1js110s9r4nkdfps27kphn65bw2pxn0
  got:    sha256:00d0wjnr5y3bl95lma8sdwvqqs7fd0k43azawp1kb29kqnrlismg
```

**Comment by novoxudonoser** [comment]
> hmm, seems a little cursed here
@Lassulus  That is really strange, I changed the sha in last commit https://github.com/NixOS/nixpkgs/pull/70615/commits/2742abcf4287a326829fef6a01b9ca15b4a252cf . Maybe you did not pull last changes, or some other human-like error ? Can we call someone 3rd to reproduce this? Btw, do you use 19.09 (updated reasonably getting some strange bugs with packets like python pip ) ?

**Comment by Lassulus** [comment]
I'm testing it with nix-review from unstable. the last hash mismatch was the source file, the new one is about the go-modules. so I guess there is another mismatch which is not apparent on your system?

**Comment by novoxudonoser** [comment]
Foud that installs not passing because of one of the dependencies no more present. The https://github.com/actions/workflow-parser with 404, seems GitHub changing their location of action repository, and code in master of act is not fixed about this. So, to pass build need to wait for act developers when the will fix dependencies. @kalbasit @Lassulus what to do with pr - close or keep open?

**Comment by Lassulus** [comment]
if its gonna be a new release, maybe its easier to close this PR and create a new PR with the new version?

**Comment by jonringer** [comment]
closed in favor of #76536

**Pull request closed without merging**